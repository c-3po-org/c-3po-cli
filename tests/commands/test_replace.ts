import * as path from "path";
import * as fs from "fs";
import { execSync } from "child_process";
import * as tmp from "tmp";

const replaceDirPath = path.resolve(__dirname, "../fixtures/replaceTest");
const poPath = path.join(replaceDirPath, "translations.po");

test("replace translations", () => {
    const tmpFolder = tmp.dirSync();
    execSync(
        `ts-node src/index.ts replace ${poPath} ${tmpFolder.name} ${
            replaceDirPath
        }`
    );
    const testFile = fs
        .readFileSync(path.join(tmpFolder.name, "test.js"))
        .toString();
    const nestedFile = fs
        .readFileSync(path.join(tmpFolder.name, "nested/test2.js"))
        .toString();
    expect(testFile).toMatchSnapshot();
    expect(nestedFile).toMatchSnapshot();
});
