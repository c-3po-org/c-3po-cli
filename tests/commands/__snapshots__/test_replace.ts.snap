// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace translations 1`] = `
"\\"use strict\\";

\\"test translation\\";"
`;

exports[`replace translations 2`] = `
"\\"use strict\\";

\\"test 2 translation\\";"
`;
